easybuild/tools/github.py

```diff
@@ -754,3 +971,71 @@ def __init__(self, user):
 def fetch_github_token(user):
     """Fetch GitHub token for specified user from keyring."""
     return GithubToken(user).token
+
+
+@only_if_module_is_available('keyring')
+def install_github_token(github_user, silent=False):
+    """
+    Install specified GitHub token for specified user.
+
+    @param github_user: GitHub user to install token for
+    @param silent: keep quiet (don't print any messages)
+    """
+    if github_user is None:
+        raise EasyBuildError("GitHub user must be specified to install GitHub token")
+
+    # check if there's a token available already
+    current_token = fetch_github_token(github_user)
+
+    if current_token:
+        current_token = '%s..%s' % (current_token[:3], current_token[-3:])
+        if build_option('force'):
+            msg = "WARNING: overwriting installed token '%s' for user '%s'..." % (current_token, github_user)
+            print_msg(msg, prefix=False, silent=silent)
+        else:
+            raise EasyBuildError("Installed token '%s' found for user '%s', not overwriting it without --force",
+                                 current_token, github_user)
+
+    # get token to install
+    token = getpass.getpass(prompt="Token: ")
+
+    # validate token before installing it
+    print_msg("Validating token...", prefix=False, silent=silent)
+    valid_token = validate_github_token(token, github_user)
+    if valid_token:
+        print_msg("Token seems to be valid, installing it.", prefix=False, silent=silent)
+    else:
+        raise EasyBuildError("Token validation failed, not installing it. Please verify your token and try again.")
+
+    # install token
+    keyring.set_password(KEYRING_GITHUB_TOKEN, github_user, token)
```

Comment:
Yes, I'll try and make that clear in the docs.

See also https://github.com/hpcugent/easybuild/wiki/Review-process-for-contributions#setting-things-up